python/cog/server/runner.py

```diff
@@ -381,16 +394,25 @@ def shutdown(self) -> None:
         if self._child.is_alive():
             self.log.info("child is alive during shutdown, sending Shutdown event")
             self._events.send(Shutdown())
+            asyncio.create_task(self.terminate_later())
+
+    async def terminate_later(self) -> None:
+        self.log.info("scheduling termination in 10s")
+        await asyncio.sleep(10)
+        self.log.info("10s has passed, terminating")
+        self.terminate()
 
     def terminate(self) -> None:
```

Comment:
Does runner even need a `terminate` method? What is the scenario in which we need it? As far as I can tell we need it to have a single `shutdown` method which waits until things have been correctly shut down. If it fails to do that (within a timeout that should be coordinated at the top of the program), we SIGKILL ourselves?

Reply:
that makes sense